### PR TITLE
[SYCL] Fix error ‘CHAR_BIT’ was not declared in this scope in sycl/source/stream.cpp

### PR DESCRIPTION
**Pull request by jchlanda**
Fix:
error: ‘CHAR_BIT’ was not declared in this scope

### CODE REVIEW

**Comment by jchlanda** [comment]
Looks like CI is experiencing some problems: https://github.com/intel/llvm/actions/runs/5794615743/job/15704584428?pr=10726#step:10:49

**Comment by jchlanda** [comment]
This was discovered by: https://github.com/intel/llvm/pull/9772

**Comment by jchlanda** [comment]
Windows runner seems to be in a broken state:
```
cmake : The term 'cmake' is not recognized as the name of a cmdlet, function, script file, or operable program. Check 
the spelling of the name, or if a path was included, verify that the path is correct and try again.
At D:\github\actions-runner\_work\_temp\0f2dcde1-b197-424a-a3f7-ea6d5d45a736.ps1:2 char:1
+ cmake --build build --target check-sycl
+ ~~~~~
    + CategoryInfo          : ObjectNotFound: (cmake:String) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : CommandNotFoundException
```

**Comment by KornevNikita** [comment]
@aelovikov-intel there is some issue with configure the job

File: sycl/source/stream.cpp [hunk 1]
```diff
@@ -6,6 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <climits>
```
Comment (bader) [inline sycl/source/stream.cpp:9]: @jchlanda, please, move this include after current includes and put empty line before `#include <climits>` to suppress clang-format re-ordering.
Including standard headers after SYCL headers allows us to detect cases when standard header includes are missing in SYCL headers.

Comment (jchlanda) [reply sycl/source/stream.cpp]: Done, thank you for explaining the reasoning.

**Review by bader — APPROVED** [review]

**Review by bso-intel — APPROVED** [review]

**Comment by jchlanda** [comment]
The failures are not related (`amdgpu`) or CI glitches (`gen12`), @intel/llvm-gatekeepers

**Pull request merged**